src/dmclock_server.h

```diff
@@ -1084,6 +1114,24 @@ namespace crimson {
 	delete_from_heap(client, limit_heap);
 	delete_from_heap(client, ready_heap);
       }
+
+
+      // data_mtx must be held by caller
+      void do_update_client_info(const C& client_id) {
+	auto client_it = client_map.find(client_id);
+	if (client_map.end() != client_it) {
+	  ClientRec& client = (*client_it->second);
+	  client.info = client_info_f(client_id);
```

Comment:
Has this code been tested? ClientInfo declares all its variables as const, and it has no assignment operator. If this is to work, I think we'll have to change ClientInfo so the variables are no longer const and there is an assignment operator.